Add `use_shallow_replication` field for Compute Gallery/SIG builds, fix replica count type (#337)
hashicorp/packer-plugin-azure: .web-docs/components/builder/arm/README.md

- `shared_gallery_image_version_end_of_life_date` (string) - The end of life date (2006-01-02T15:04:05.99Z) of the gallery Image Version. This property
  can be used for decommissioning purposes.

- `shared_image_gallery_replica_count` (int64) - The number of replicas of the Image Version to be created per region.
  Replica count must be between 1 and 100, but 50 replicas should be sufficient for most use cases.

>   This value can only be 1 when using shallow replication

Reviewer: ```suggestion
  When using shallow replication `use_shallow_replication=true` this value must be set to 1.
```

Author: I will re-run make generate to create this suggestion from the config changes
